src/scala/ly/stealth/mesos/kafka/Scheduler.scala

```diff
@@ -87,8 +87,14 @@ object Scheduler extends org.apache.mesos.Scheduler {
         "host.name" -> offer.getHostname
       )
 
-      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0)
-        defaults += ("listeners" -> s"PLAINTEXT://:${reservation.port}")
+      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0) {
+        if (broker.bindAddress != null) {
+          defaults += ("listeners" -> s"PLAINTEXT://${offer.getHostname}:${reservation.port}")
```

Comment:
I think you should be using the result of bindAddress.resolve() here for the listener address rather than the offer, since the bindAddress could be a different interface than the slave process is configured to advertise.

Once you change it, you'll have to then move this logic into the BrokerServer code so it runs on the slave itself and can correctly resolve the address.  This logic currently exists in Broker.options(...) to set the host.name property correctly on the slave.